### DISCUSSION TITLE
我们的仓库 160 爱慕了

### DISCUSSION THREAD

**Discussion by Arondight** [Arondight/Adachi-BOT#555] [General]
我感觉七国开完，有可能上 300 爱慕了，其实最多的还是字体占用太大了，不过似乎直接放在仓库里，要比 git-lfs 好一些，后者会让部署的流程变麻烦，希望有个好用的免费代理吧，要不然其他人克隆项目有点痛苦

```
Processing blobs: 4330
Processing trees: 4580
Processing commits: 1221
Matching commits to trees: 1221
Processing annotated tags: 2
Processing references: 8
| Name                         | Value     | Level of concern               |
| ---------------------------- | --------- | ------------------------------ |
| Overall repository size      |           |                                |
| * Commits                    |           |                                |
|   * Count                    |  1.22 k   |                                |
|   * Total size               |   748 KiB |                                |
| * Trees                      |           |                                |
|   * Count                    |  4.58 k   |                                |
|   * Total size               |  1.79 MiB |                                |
|   * Total tree entries       |  49.1 k   |                                |
| * Blobs                      |           |                                |
|   * Count                    |  4.33 k   |                                |
|   * Total size               |   225 MiB |                                |
| * Annotated tags             |           |                                |
|   * Count                    |     2     |                                |
| * References                 |           |                                |
|   * Count                    |     8     |                                |
|                              |           |                                |
| Biggest objects              |           |                                |
| * Commits                    |           |                                |
|   * Maximum size         [1] |  1.69 KiB |                                |
|   * Maximum parents      [2] |     2     |                                |
| * Trees                      |           |                                |
|   * Maximum entries      [3] |   200     |                                |
| * Blobs                      |           |                                |
|   * Maximum size         [4] |  16.6 MiB | *                              |
|                              |           |                                |
| History structure            |           |                                |
| * Maximum history depth      |  1.07 k   |                                |
| * Maximum tag depth      [5] |     1     |                                |
|                              |           |                                |
| Biggest checkouts            |           |                                |
| * Number of directories  [6] |   119     |                                |
| * Maximum path depth     [6] |     5     |                                |
| * Maximum path length    [6] |    59 B   |                                |
| * Number of files        [6] |  1.44 k   |                                |
| * Total size of files    [7] |   160 MiB |                                |
| * Number of symlinks         |     0     |                                |
| * Number of submodules       |     0     |                                |

[1]  07f31d3ab7f8791d78e539dba6ad378238972fd6
[2]  2deed37cd17aa8b2380027b40c28225fea70a977
[3]  385efed0a4c420338152d079337b22bf79719e62 (refs/heads/dev:resources/Version2/info/image)
[4]  13d0fee3a6a982b0dccef570fa5e968899d265e9 (refs/heads/dev:data/font/YunXiJinShu.ttf)
[5]  962da1a9c8d63fb62b37492ecbb22edec70fdbdf (refs/tags/v1.5.0)
[6]  801aadcb9a783ad77936f352b0ce10c71060bfb7 (refs/heads/dev^{tree})
[7]  5a64bf65870db7f82150d8c12e1b74331df351dd (2deed37cd17aa8b2380027b40c28225fea70a977^{tree})
```

**Comment by mark9804** [comment]
国内好像有一个代理项目叫 fastgit 吧，我没有国内的机器，不知道速度如何

**Reply by Arondight** [reply]
这个用起来感觉很不稳定，目前没什么好办法

**Reply by Xm798** [reply]
> 这个用起来感觉很不稳定，目前没什么好办法

自己搞了个 CF Workers 在用，还可以。每天有 100k 的请求数上限，小范围用用够用了。

**Reply by mark9804** [reply]
[仓库镜像管理 （ Gitee <-> Github 双向同步） - Gitee.com](https://gitee.com/help/articles/4336#article-header8)

这个呢

**Reply by Xm798** [reply]
> [仓库镜像管理 （ Gitee <-> Github 双向同步） - Gitee.com](https://gitee.com/help/articles/4336#article-header8)
> 
> 这个呢

这个可以 但是有延迟 不是实时同步的

**Reply by Arondight** [reply]
呃，不知道你俩能同步么，我的 gitee 上不想放这个项目，被同事看到了不太好

**Reply by mark9804** [reply]
> 呃，不知道你俩能同步么，我的 gitee 上不想放这个项目，被同事看到了不太好

刚刚看了一下不是所有仓库都能开通的，这个仓库至少不行

~~我刚刚看到一个 gitclone.com , 不知道行不行~~ 感觉不如 fastgit ……速度

**Reply by Arondight** [reply]
我感觉找个什么地方挂个脚本应该也行吧，例如每五分钟 git pull 一次然后 git push ，不过我的 gitee 实在不想跟 github 联系在一起，有点麻烦，这个东西挂 action 铁定是违反规则，或者等等 gitee 看看后续做成什么样子，看样子这个同步因为还在内测所以没完全开放

**Reply by mark9804** [reply]
```
git clone https://github.com.cnpmjs.org/Arondight/Adachi-BOT.git
```

```
Cloning into 'Adachi-BOT'...
remote: Enumerating objects: 11884, done.
remote: Counting objects: 100% (1484/1484), done.
remote: Compressing objects: 100% (479/479), done.
remote: Total 11884 (delta 1163), reused 1139 (delta 1002), pack-reused 10400
Receiving objects: 100% (11884/11884), 187.52 MiB | 11.86 MiB/s, done.
Resolving deltas: 100% (6395/6395), done.
Updating files: 100% (2080/2080), done.
```

**Reply by Arondight** [reply]
这个我试过了，效果不如 fastgit 好，因为这个网站时常开摆

**Reply by mark9804** [reply]
那就没有好的解决方法了，我们也不应该给某个镜像做背书说推荐用这个，把这个问题留给用户吧

**Reply by Xm798** [reply]
> 我感觉找个什么地方挂个脚本应该也行吧，例如每五分钟 git pull 一次然后 git push ，不过我的 gitee 实在不想跟 github 联系在一起，有点麻烦，这个东西挂 action 铁定是违反规则，或者等等 gitee 看看后续做成什么样子，看样子这个同步因为还在内测所以没完全开放

这个用action违反规则吗？我倒是看到一个action，貌似用的人蛮多的 https://github.com/Yikun/hub-mirror-action

**Reply by Arondight** [reply]
这个好像还不错，我明天上班想想，最后一天应该不会太忙，我看看怎么把我的 gitee 对同事变透明，或者借个号用

**Reply by mark9804** [reply]
老大哥还是不要拿自己账户冒险吧，我的 gitee 账号用都没用，也不怕社死，~~也不怕律师函，~~ 可以用我的

**Reply by Arondight** [reply]
只能用我的吧，用你的最后还是手动更新不是么，我感觉你手动 merge 这个仓库到你的 github 然后自动同步到 gitee，和手动 merge 到 gitee 没有什么本质的区别……

**Reply by Xm798** [reply]
我在本地搞了个脚本 15min检查一次仓库 自动同步到gitee
镜像：https://gitee.com/Xm798/Adachi-BOT
（我香港的服务器终于派上用场了

**Reply by Arondight** [reply]
太强了，顺便改一下 readme 吧，把那个“使用代理克隆本项目”改一下描述和地址，例如“使用镜像仓库”之类的，然后贴个 gitee 上去，我刚回来没多久体力还没打完，实在不想开虚拟机了……

**Reply by mark9804** [reply]
> 我在本地搞了个脚本 15min检查一次仓库 自动同步到gitee 镜像：https://gitee.com/Xm798/Adachi-BOT （我香港的服务器终于派上用场了

我草，太强了

**Reply by mark9804** [reply]
等一下，码云会自动把 GitHub 账号的提交和 gitee 账号关联起来，这个不知道有没有影响

**Reply by Arondight** [reply]
> 等一下，码云会自动把 GitHub 账号的提交和 gitee 账号关联起来，这个不知道有没有影响

没有

**Reply by Xm798** [reply]
> 等一下，码云会自动把 GitHub 账号的提交和 gitee 账号关联起来，这个不知道有没有影响

这个貌似是邮箱的原因…我一直用匿名邮箱，所以匹配不到hhhh

**Reply by Arondight** [reply]
我主要是同事看到我上班时间居然有啥都不相干的项目的提交会很尴尬，虽然都不怎么用 gitee 不过还是不要无事生非了，特别是万一传到领导耳朵里不太好，项目没挂在我账号下面就无所谓

**Reply by Xm798** [reply]
刚才学习到了git lfs的新知识，真不错啊。

可惜gitee的lfs只对付费企业开放，真不太行。

**Reply by Arondight** [reply]
git lfs 我本来想在这个项目中使用的，但是对于绝大多数发行版来说 git lfs 需要额外安装软件包，并且 pull 会变麻烦，我感觉大部分 git 都没怎么用过的实在不好上手这个东西，就给毙了，所有的二进制文件都直接提交了

**Reply by mark9804** [reply]
如果用 lfs 的话，我们的 invalid issue数量可能会指数级上升的，keep it simple

**Comment by Xm798** [comment]
说起来Gitee真的好难用 随便点个什么就让登录 页面乱的啥都找不到……

**Reply by Arondight** [reply]
大概是真的自己开发的，没用 gitlab 换皮

**Comment by Arondight** [comment]
兄弟们，我找到完美的解决方案了！

https://gitcode.net/iSpeller/Adachi-BOT.git

**Reply by Arondight** [reply]
镜像自动同步，wiki 也同步过去了，太良心了

**Reply by mark9804** [reply]
草，确实完美

**Reply by Xm798** [reply]
好良心啊

**Reply by Xm798** [reply]
想不到竟然是CSDN的………………………………

**Reply by Arondight** [reply]
> 好良心啊

分支，tag，wiki，release，全都过去了，我的 token 里面只开了三个只读权限，我甚至感觉给足了权限会过去更多东西

**Comment by Arondight** [comment]
各位，我发现楼上那个完美的解决方案不是很完美，但是可以处，之所以不完美是因为[这个](https://gitcode.net/gitcode/help-docs/-/issues/28)，之所以说他可以处是因为……

![图片](https://user-images.githubusercontent.com/13673702/156868272-a888d0c4-ae22-48ae-a14b-00f884738cfa.png)

**Reply by Xm798** [reply]
手册是这么说的

>### 工作机制
为仓库启用拉取镜像功能后，会将仓库添加到队列中。每分钟一次，Sidekiq cron 作业基于以下内容计划仓库镜像的更新：
* 可用容量，这由 Sidekiq 设置确定， GitCode 参考了 [Gitlab.com](https://docs.gitlab.com/13.2/ee/user/gitlab_com/index.html#sidekiq) 的默认配置。
* 队列中已经要更新的仓库镜像数，到期时间取决于仓库镜像的最后更新时间以及重试次数。
Sidekiq 可用时，它就会开始处理仓库镜像并对它们进行拉取更新。更新仓库镜像的过程为：
* 成功后，至少需要等待 30 分钟，更新才会重新加入队列
* 失败（例如，分支与上游分支出现分叉），稍后将再次尝试。镜像最多可以发生 14 次故障，然后才不会再次进入队列进行更新[](https://github.com/Arondight/Adachi-BOT/discussions/555#how-it-works)工作机制
为仓库启用拉取镜像功能后，会将仓库添加到队列中。每分钟一次，Sidekiq cron 作业基于以下内容计划仓库镜像的更新：
可用容量，这由 Sidekiq 设置确定， GitCode 参考了 [Gitlab.com](https://docs.gitlab.com/13.2/ee/user/gitlab_com/index.html#sidekiq) 的默认配置。
队列中已经要更新的仓库镜像数，到期时间取决于仓库镜像的最后更新时间以及重试次数。
Sidekiq 可用时，它就会开始处理仓库镜像并对它们进行拉取更新。更新仓库镜像的过程为：
成功后，至少需要等待 30 分钟，更新才会重新加入队列
失败（例如，分支与上游分支出现分叉），稍后将再次尝试。镜像最多可以发生 14 次故障，然后才不会再次进入队列进行更新

**Reply by Arondight** [reply]
不知道为什么我配置的就不行，我已经把 curl 加到 cron 里了，不过还是希望能有个全自动的

**Reply by mark9804** [reply]
为啥说能处是因为这张图啊，没太看懂，`302 重定向` ，然后可以被复制为 POSIX curl ，是说只要请求该文件就会立即触发拉取吗

**Reply by Arondight** [reply]
> 为啥说能处是因为这张图啊，没太看懂，`302 重定向` ，然后可以被复制为 POSIX curl ，是说只要请求该文件就会立即触发拉取吗

复制成了一个 linux 下的 curl 命令，运行一下这个命令就同步一次，我就给加到 cron 里了，我准备看看 gitcode 有没有 ci，然后让他自己鞭策自己

**Reply by Arondight** [reply]
> 我准备看看 gitcode 有没有 ci，然后让他自己鞭策自己

gitcode 的 ci 是禁用的，需要实名上网去申请，感觉不太行，还是在 vps 上挂 cron 吧

**Comment by Arondight** [comment]
我发现 GitCode 的镜像仓库又出了个大问题，设置拉取之后，点强制更新会一直转菊花，需要复制出 删镜像仓库、重加镜像仓库 的过程来作为定时任务取代之前的点更新按钮的过程

**Reply by BnqDzj** [reply]
代码仓库定期同步好像只向入驻GitCode的企业（社区）开放

**Reply by Xm798** [reply]
> 代码仓库定期同步好像只向入驻GitCode的企业（社区）开放

极狐目前可以实现仓库镜像
